src/plugins/solvers/linear_solver.cpp

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2018 Shahrzad Shirzad
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <phylanx/config.hpp>
+#include <phylanx/ir/node_data.hpp>
+#include <phylanx/plugins/solvers/linear_solver.hpp>
+
+#include <hpx/include/lcos.hpp>
+#include <hpx/include/naming.hpp>
+#include <hpx/include/util.hpp>
+#include <hpx/throw_exception.hpp>
+#include <hpx/util/assert.hpp>
+
+#include <cmath>
+#include <cstddef>
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <blaze/Math.h>
+
+///////////////////////////////////////////////////////////////////////////////
+namespace phylanx { namespace execution_tree { namespace primitives
+{
+///////////////////////////////////////////////////////////////////////////////
+#define PHYLANX_LIN_MATCH_DATA(name)                                           \
+    hpx::util::make_tuple(name, std::vector<std::string>{name "(_1, _2)"},     \
+        &create_linear_solver, &create_primitive<linear_solver>)
+    /**/
+
+    std::vector<match_pattern_type> const linear_solver::match_data = {
+        PHYLANX_LIN_MATCH_DATA("LinSolv_LU"),
+        PHYLANX_LIN_MATCH_DATA("LinSolv_LDLT_U"),
+        PHYLANX_LIN_MATCH_DATA("LinSolv_LDLT_L")};
+
+#undef PHYLANX_LIN_MATCH_DATA
+
+    ///////////////////////////////////////////////////////////////////////////
+    linear_solver::vector_function_ptr linear_solver::get_lin_solver_map(
+        std::string const& name) const
+    {
+        static std::map<std::string, vector_function_ptr> lin_solver = {
+            {"LinSolv_LU",
```

Comment:
Should we start using CamelCase names for primitives? So far we have been using `gnu_style_names`.